### ISSUE TITLE
'concept' and 'requires' are keywords in c++2a

### ISSUE THREAD

**Issue by palebedev** [ldionne/dyno#56]
This makes the library unusable with, for example, `clang++ -std=c++2a`.

I have a local patch that appends underscores to these names, but I'm not sure how much to change in the documentation comments.

**Comment by ricejasonf** [comment]
```
/usr/local/src/emscripten/system/include/dyno/concept.hpp:27:1: error: declaration of anonymous struct
      must be a definition
struct concept;
```

@palebedev Do you still have that patch?

**Comment by ricejasonf** [comment]
I made a PR for this https://github.com/ldionne/dyno/pull/58

**Comment by palebedev** [comment]
No, but it was pretty much your PR sans documentation updates.

**Comment by ldionne** [comment]
Fixed by #58.
